### DISCUSSION TITLE
ValueObservation not triggering when tracking multiple regions

### DISCUSSION THREAD

**Discussion by yqiang** [groue/GRDB.swift#1548] [General]
Hi,
My goal is to observe the `modificationDate` column of two separate tables, and run some code when either's modificationDate is updated. The code that sets up the observation looks like this:

```
private func setupUserObservation() {
    let userRegion = UserModel.select(UserModel.Columns.modificationDate)
    let weightGoalRegion = WeightGoalModel.select(WeightGoalModel.Columns.modificationDate)

    let userObservation = ValueObservation
        .tracking(regions: [userRegion, weightGoalRegion]) { db -> [UserInfo] in
            try UserModel
                .filter(UserModel.Columns.syncDate <= UserModel.Columns.modificationDate)
                .including(required: UserModel.weightGoal)
                .asRequest(of: UserInfo.self)
                .fetchAll(db)
        }
        .removeDuplicates()

    userCancellable = userObservation.start(
        in: appDatabase.reader,
        scheduling: .async(onQueue: syncManagerQueue),
        onError: { log.error("\($0.localizedDescription)") },
        onChange: { [weak self] users in
            guard let self else { return }
            log.info("triggering sync for user")
            
            Task {
               // do something with the new users array
            }
        }
    )
}
```

When I update the `modificationDate` of a `WeightGoalModel` in a separate part of the app, I would expect this ValueObservation to be triggered, but that is not what I'm observing. Here's how I'm updating the `modificationDate`:

```
try appDatabase.writer.write { [weak self] db in
    guard let self else { return }

    energyGoal.modificationDate = try db.transactionDate
    weightGoal.modificationDate = try db.transactionDate

    try energyGoal.save(db)
    try weightGoal.save(db)
}
```

Is there something obvious I'm overlooking, or am I misunderstanding on tracked regions work?

**Comment by groue** [comment]
Hello @yqiang,

There's nothing obviously wrong in your sample code. I suggest you try the troubleshooting techniques described in the FAQ: [Why is ValueObservation not publishing value changes?](https://github.com/groue/GRDB.swift/blob/master/README.md#why-is-valueobservation-not-publishing-value-changes)

Please report your findings!

**Reply by yqiang** [reply]
Thank you for pointing me to the documentation - not sure why I didn't consult it first this time :) 

The debug logging was extremely helpful. It turns out that GRDB was behaving as expected, but because I was updating the the modificationDate on `weightGoal` and not the `UserModel`, this line filtered out the results and returned an empty array.

`.filter(UserModel.Columns.syncDate <= UserModel.Columns.modificationDate)`

If i also update the `modificationDate` of the `UserModel` when I update the `WeightGoalModel`, everything works as expected. 

Thanks for the helpful & prompt reply!

**Reply by groue** [reply]
Glad you found your solution! Happy GRDB!
